node.go

```diff
@@ -136,21 +140,76 @@ func (n *Node) EnableMDNSDiscovery(notifee discovery.Notifee, interval time.Dura
 }
 
 // NewNode creates a new ipfs basichost node with given identity
-func NewNode(listenAddr string, protoMux string, agent *LibP2PAgent) (node *Node, err error) {
+func NewNode(listenPort int, protoMux string, agent *LibP2PAgent) (node *Node, err error) {
 	Debugf("Creating new node with protoMux: %s\n", protoMux)
 	nodeID, _, err := agent.NodeID()
 	if err != nil {
 		return
 	}
 
 	var n Node
+	listenAddr := fmt.Sprintf("/ip4/0.0.0.0/tcp/%d", listenPort)
```

Comment:
Not so sure about moving the libp2p listenAddr into NewNode.  I was assuming we wanted to allow all the transports that libp2p offers, maybe we should just refactor that Sprintf into something like "makeIPv4ListenAddr" or something...  Also, for the tests I use 127.0.0.1.  I'm getting this error during test time with 0.0.0.0:

```
 * /home/eric/go/src/github.com/metacurrency/holochain/node_test.go 
  Line 89:
  Expected: nil
  Actual:   'dial attempt failed: <peer.ID NN6oDi> --> <peer.ID XJphwW> dial attempt failed: Attempted to connect to zero address: /ip4/0.0.0.0/tcp/4321'
```